client/src/language/RequestManager.ts

```diff
@@ -18,6 +18,14 @@ export class RequestManager {
     const params: RequestParams['EmbeddedLanguageTypeOnPosition'] = { uriString, position }
     return await this.client?.sendRequest(RequestMethod.EmbeddedLanguageTypeOnPosition, params)
   }
+
+  checkIsSymbolDefinedInRecipe = async (
+    recipeName: string,
+    symbolName: string
+  ): RequestResult['IsSymbolDefinedInRecipe'] => {
+    const params: RequestParams['IsSymbolDefinedInRecipe'] = { recipeName, symbolName }
```

Comment:
In order to reduce the number of requests to the server, I suggest doing one request per file analyzing where you request ALL the symbols from the server. Just get the whole array from the server, then do what you need with the variables you are concerned with.
That way we can also use that request for other purposes later. We already have getVariable() for just one variable.

Reply:
Also, move the code for the request near getVariableValue() (languageClient.ts) for proper code grouping. The new function could be `getAllVariableValues(client,recipe): LastScanResult|undefined`.
You could rename that type to RecipeScanResult in a refactor commit.

Reply:
Done